### ISSUE TITLE
option to read capitalized letters individually, as words or let synthesizer handle it

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4520]
**Reported by blindbhavya on 2014-10-03 08:20**
I propose a combo box in NVDA to read capitalized letters with the following options
individual = this option instructs the synthesizer to read capitalized letters individually for e.g. NVDA will be read as n v d a
words = This option instructs the synthesizer to read capitalized letters as words for e.g. UNESCO will be read as unesco
let synthesizer handle it = This option allows the synthesizer to handle the reading of capitalized letters (which is what happens currently).
Hope this may be implemented.

**Comment by bhavyashah** [comment]
A friendly ping in case any developer wants to give this feature request a try or @feerrenrut would like to assign this a priority.

**Comment by feerrenrut** [comment]
Personally I'm not convinced that this is something that should be solved within NVDA. It would be much better if the synth was responsible for this, I think bugs should filed against synths that don't get this right. That said, if the same synth produces better results with other screen readers, then perhaps there is more that we can do.

**Comment by vimalan-sakthivel** [comment]
NVDA (version 2.17.2) reads "NEW" as N.E.W.

**Comment by feerrenrut** [comment]
@vimalan-sakthivel which synthesizer were you using? Espeak? Does this happen with other synthesizers?

When I test "NEW" with NVDA Version: `master-14328,c6379eb3` (close to the upcoming release of NVDA 2017.3) using espeak I get the described behaviour. However when I use the Microsoft Speech API version 5 synthesizer "NEW" is spoken as "new". 

@vimalan-sakthivel could you please create a new issue describing this on the espeak-ng repository: https://github.com/espeak-ng/espeak-ng/issues

Closing this issue since this seems like synth specific behaviour.

**Comment by bhavyashah** [comment]
@feerrenrut I would like to kindly request reopening this issue. This ticket does not deal with synth specific pronunciation bugs but with a feature request for a combo box to toggle whether capitalized words should be read individually, e.g. N V D A, as complete words, e.g. nvda, or to allow the synthesizer to handle it via its own pronunciation rules.

**Comment by feerrenrut** [comment]
You're right, I didn't properly consider the initial request. But I don't think it really changes the outcome. To start with, I would categorise this as a new feature rather than a bug.  As a feature, its not well defined enough, in terms of the problem it's trying to solve, and the value it would bring to users.

To start with I think its important for this issue to distinguish between an (pronounceable) acronym and an (unpronounceable) initialism. See https://www.dailywritingtips.com/acronym-vs-initialism/

I expect the process of trying to define this better (through triage) to be prohibitively expensive. So rather than leaving it open, I still think it's best to close the issue. 
If you wish to persevere with this issue, I suggest first defining a set of concrete use cases, and trying to define general users stories for this. Also consider the situations that will be problematic (eg math in lower case, documents / titles / in upper case). 

Even so it's not clear that we could come up with an implementation that would improve the situation from within NVDA.

Given the following example concrete use-cases
Acronym, pronounce by word:
- UNESCO

Initialism, pronounce by letter:
- NVDA
- XYZ (in the context of math)
- OPQ (in the context of math)

leads me to the following questions / thoughts:
- What about (in the context of math) xyz (in lower case)?
- Applying a rule based only on the upper/lower case-ness of the word will result in lots of edge cases that are not addressed (or made worse).
- Having an option to swap between two outcomes (letter pronunciation / word pronunciation) complicates the usage of NVDA. Neither are likely to result in problem free behaviour in any single use case. This adds the complication of user configurable behaviour, without providing a concrete solution.
- Adding to the rule a consideration for the length of the word mitigate the impact of some errors (making them less inconvenient). It would take less time to incorrectly read out the individual letters of STOP than reading the individual letters of STATIONARY for instance. There are still many edge cases, there are likely long initialisms.
- It seems like the key factor on whether or not something should be said as a word, or as a set of letters, is based on whether it is considered pronounceable. This is still a fuzzy concept, the combination "xyz" could be considered pronounceable. In terms of knowing whether something is pronounceable, the synthesizer is in the best position to have the information to decide on that.

I still think that the best place to push for a fix to this is with the synthesizer. Unfortunately that means that it likely needs to be fixed by several synthesizers. I would suggest that the synthesizer pronounce letters in any cases where there is some doubt that the word is pronounceable, for instance "xyz". I would add an exception to this, if the word is longer than a certain limit (maybe 8 characters) then pronounce it as a word. Because a really long initialism is tedious to hear letter by letter, and when encountering nonsense its better for it to be over sooner. The user can always step through character by character to try to make sense of it.

**Comment by bhavyashah** [comment]
The reasoning you provide is very cogent. Thus, I guess the current status (closed) of this ticket is probably justified. It was interesting however to understand how any feature needs to be thoroughly evaluated from all angles, use cases, etc. Thanks @feerrenrut.

**Comment by vimalan-sakthivel** [comment]
@feerrenrut 
you're correct. I used eSpeak NG. and like you mentioned this issue was not observed with Microsoft Speech API version 5. I have raised a new issue with eSpeak NG. Thank you! 
https://github.com/espeak-ng/espeak-ng/issues/301
